packages/lesswrong/components/users/EmailTokenPage.jsx

```diff
@@ -0,0 +1,37 @@
+import React, {Component} from 'react';
+import { Components, registerComponent, withMutation } from 'meteor/vulcan:core';
+import { getComponent } from 'meteor/vulcan:lib';
+
+class EmailTokenPage extends Component
+{
+  state = {
+    loading: true,
+    useTokenResult: null
+  };
+  
+  componentDidMount() {
+    const { params: { token } } = this.props;
+    this.props.useEmailToken({token}).then((mutationResult) => {
```

Comment:
This was to avoid making `componentDidMount` async, since that's not the type signature React is expecting. Some research suggests that marking `componentDidMount` async is probably harmless (the promise it returns will just be ignored), but it seems better to be explicit about the fact that this is splitting off a background thread and not being awaited.